src/Sarif.UnitTests/Core/StackTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Linq;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    [TestClass]
+    public class StackTests
+    {
+        [TestMethod]
+        public void Stack_CreateFromStackTrace()
+        {
+            var dotNetStack = new StackTrace();
+            Stack stack = Stack.Create(dotNetStack);
+
+            // The .NET StackTrace.ToString() override must preserve a trailing NewLine
+            // for compatibility reasons. We do not retain this behavior in ToString()
+            // but provide a facility for adding the trailing NewLine
+            Assert.AreEqual(dotNetStack.ToString(), stack.ToString(StackFormat.TrailingNewLine));
```

Comment:
I am ripe for convincing. However, fluent assertions fail in one basic case, they convert Environment.NewLine to escaped char sequences (to render output on a single line). This tanks easy comparison of stack ToString(). So I reverted to Assert in this case. I have other examples.

IOW while the readability is interesting and powerful, the experience grokking failures when you examine output text in VS is inferior to MSTEST API. in my experience to date.